cpp/src/aztec3/circuits/abis/c_bind.cpp

```diff
@@ -146,6 +147,24 @@ WASM_EXPORT void abis__compute_function_selector(char const* func_sig_cstr, uint
     std::copy_n(hash_bytes, aztec3::FUNCTION_SELECTOR_NUM_BYTES, output);
 }
 
+/**
+ * @brief Hash/compress verification key data.
+ * This is a WASM-export that can be called from Typescript.
+ *
+ * @details Pedersen compress VK to use later when computing function leaf
+ * or constructor hash. Return the serialized results in the `output` buffer.
+ *
+ * @param vk_data_buf buffer of bytes representing serialized verification_key_data
+ * @param output buffer that will contain the output. The serialized vk_hash.
+ */
+WASM_EXPORT void abis__hash_vk(uint8_t const* vk_data_buf, uint8_t* output)
+{
```

Comment:
Hmm, I suppose the typescript is just expected to pass a buffer of the right size? Have we considered bbmalloc'ing a returned buffer so typescript just frees it?

Reply:
Hmm I like this idea. Since I do this in all of the cbinds I've written, I will submit a single PR after these that bbmallocs all of the output buffers.

Reply:
Good with me, yeah I wasn't too keen on making you redo all these PRs